wiki/beginners.md

```diff
@@ -0,0 +1,450 @@
+
+``` javascript
+require('remotestoragejs/release/stable/remotestorage.amd.js');
+// window.RemoteStorage is now available
+```
+
+You also need to define a shim for `XMLHttpRequest` such that Webpack
+won't be looking for it. In your `webpack.config.js`:
+
+``` javascript
+var webpack = require("webpack");
+  
+module.exports = {
+  externals: [
+    {xmlhttprequest: "{XMLHttpRequest:XMLHttpRequest}"}
+  ],
+}
+```
+
+We know that this sort of usage is not very aesthetic and are working on
+fixing it.
+
+#### Git
+
+` git clone `<https://github.com/remotestorage/remotestorage.js.git>
+
+You can find the release builds in the `release/` dir.
+
+#### Direct Download
+
+The latest stable release files can be downloaded [directly from our
+GitHub
+repository](https://github.com/remotestorage/remotestorage.js/tree/master/release/stable).
+
+You can download an archive of the whole repository from the [releases
+page](https://github.com/remotestorage/remotestorage.js/releases),
+unpack it on your computer, and find the release builds in
+`release/stable/`.
+
+#### CDN
+
+` `<http://cdnjs.com/libraries/remoteStorage/>
+
+Content Delivery Network (CDN) provided by cdnjs.com.
+
+### Add a data module
+
+The remoteStorage [ protocol](Protocol "wikilink") requires apps to be
+authorized for one or more base directories in order to read and write
+data. These are called categories, and they provide the scope for both
+per-app as well as common data.
+
+The remoteStorage community aims to create standard formats for objects
+stored in common categories, in order to make stored data
+interchangeable between applications. For example a to-do list app could
+access the same task list as a time tracking app. Both would claim
+access to the "tasks" category and manipulate "todo" objects in folders
+therein.
+
+remoteStorage.js uses per-category [ data
+modules](Data_Modules "wikilink"), which provide functionality for
+manipulating data within their scope (e.g. "documents", "contacts",
+"bookmarks"). You can choose from a variety of common modules that
+already exist, or build your own.
+
+#### Using an existing data module
+
+Existing data modules are found in the [modules
+repository](https://github.com/remotestorage/modules), under
+[`dist/`](https://github.com/remotestorage/modules/dist). Their
+documentation can be found in the [modules
+docs](https://remotestorage.github.io/modules). *Beware that most common
+data modules are still under development.*
+
+If there's no data module that fits your data, you can either extend one
+or write your own — and hopefully propose a new common module.
+
+##### Example
+
+This example uses the [bookmarks
+module](https://github.com/remotestorage/modules/blob/master/dist/bookmarks.js).
+You may find helpful to take a look at the [bookmarks module
+docs](http://remotestorage.github.io/modules/files/bookmarks-js.html).
+
+First, we import `remotestorage.js` and our chosen module
+`remotestorage-bookmarks.js`.
+
+``` html5
```

Comment:
This code block renders oddly enough to highlight it here. `html5` may not be a valid syntax highlighter for Jekyll. Try using `html` without the space to the annotation.